src/transport/mqtt/connection.rs

```diff
@@ -364,7 +364,23 @@ impl Disconnected {
     /// session. If it fails, it returns an error so that the whole connection process can
     /// be retried.
     async fn reconnect(&mut self, conn: &mut Connection, client_id: ClientId<&str>) -> Next {
-        let api = ApiClient::from_transport(&conn.provider, client_id.realm, client_id.device_id);
+        // retrieve the client config options
+        let tls_cfg = match conn.provider.api_tls_config().await {
+            Ok(cfg) => cfg,
+            Err(err) => {
+                error!(error = %Report::new(err),"couldn't pair device");
+
+                return Next::Same;
+            }
+        };
```

Comment:
I'm not too sure with this change, I'm worried this will create an infinite loop.

Reply:
I think the best course of action is that the transport provider should read the root cert store in creation, so we can move the error during configuration and not risk looping indefinitely during reconnection.